adoc-mode.el

```diff
@@ -1991,6 +1991,20 @@ LANG is a string, and the returned major mode is a symbol."
          (intern (concat lang "-mode"))
          (intern (concat (downcase lang) "-mode")))))
 
+(defun adoc-map-intervals (fun property &optional beg end object)
+  "Apply FUN to all intervals of PROPERTY in OBJECT in the region from BEG to END.
+FUN is called with two arguments: the beginning and the end of the interval."
+  (unless object (setq object (current-buffer)))
```

Comment:
Why don't you just use `let` and something like `(or beg (...))`. Using `setq` to establish initial bindings is quite uncommon.

Reply:
There are many techniques for assigning default arguments.
The `unless ... setq`-Method is one of the usual ones. Examples:
- [`string` of `put-image`](https://git.savannah.gnu.org/cgit/emacs.git/tree/lisp/image.el#n610)
- [`name` of `abbrev-insert](https://git.savannah.gnu.org/cgit/emacs.git/tree/lisp/abbrev.el#n847)
- [`buffer` of `apropos-local-variable`](https://git.savannah.gnu.org/cgit/emacs.git/tree/lisp/apropos.el#n545)
- and many more...

It works well since function arguments are lexically bound args and setq is a special form that can handle lexically bound args. The quote in `(setq sym val)` is not really evaluated in the sense of `(set 'sym val)` which couldn't handle lexically bound args well.